lib/engine/gameutils/undo_helper.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+module Engine
+  module GameUtils
+    class UndoHelper
+      attr_reader :undo_list
+
+      def initialize(actions = nil)
+        process_actions(actions)
+      end
+
+      def process_actions(actions)
+        @undo_list = []
+        @last_known_undo = 0
+        return if actions.nil?
+
+        # List of actions that undo does not apply to
+        kept_actions = Action.constants.map do |c|
```

Comment:
hm maybe too much magic, maybe we can just stick to hardcoding

Reply:
keep_actions=['message'] okay?